test/blackbox-tests/test-cases/cram/git-diff-fail.t

```diff
@@ -15,8 +15,10 @@ First we make a cram test:
 We need to avoid the special treatment of the test when INSIDE_DUNE is set:
   $ unset INSIDE_DUNE
 
-We get nonsense internal output in the display:
-
-  $ dune build --root=. --diff-command="exit 1" --display=short @runtest 2>&1 >/dev/null | head -n2
+We do not observe the leaking of the display of internal processes when running
+dune build. Note that we scrub the actual reported error due to the build tool
+being bogus.
+  $ dune build --root=. --diff-command="exit 1; echo" --display=short @runtest | sed '/^(cd/d' 
```

Comment:
@rgrinberg I can't seem to work out how to sanitize the output here. Could you have a go?

Reply:
@Alizter I pushed something. There are several things going on here:
- you're not seeing the `cd` locally because they rely on `inside_ci`. There's an example of that in the `print-diff.t` test. I added that in order to have a suppressable output locally
- the `cd` message is on stderr, so it does not go through the pipe
- redirecting stderr to stdout is not enough, because the pipe will "absorb" the error code and `[1]` is not printed anymore. `bash` is necessary to fix that unfortunately.

Reply:
Thanks!